packages/core/test/insured-bridge/L2_DepositBox.js

```diff
@@ -0,0 +1,52 @@
+const hre = require("hardhat");
+const { didContractThrow } = require("@uma/common");
+const { getContract } = hre;
+
+const { assert } = require("chai");
+
+const { deployOptimismContractMock } = require("./helpers/SmockitHelper");
+
+// Tested contract
+const BridgeDepositBox = getContract("BridgeDepositBox");
+
+// Contract objects
+let depositBox;
+let l2CrossDomainMessengerMock;
+
+describe("L2_depositBox", () => {
+  let accounts, deployer, user1, l1Owner, l2MessengerImpersonator, rando;
+
+  beforeEach(async function () {
+    accounts = await web3.eth.getAccounts();
+    [deployer, user1, l1Owner, l2MessengerImpersonator, rando] = accounts;
+
+    l2CrossDomainMessengerMock = await deployOptimismContractMock("OVM_L2CrossDomainMessenger", {
+      address: l2MessengerImpersonator,
+    });
+
+    depositBox = await BridgeDepositBox.new(l2CrossDomainMessengerMock.options.address, l1Owner).send({
+      from: deployer,
+    });
+  });
+  describe("Box Ownership logic", () => {
+    it("Transfer ownership", async () => {
+      // Owner should start out as the set owner.
+      assert.equal(await depositBox.methods.l1Owner().call(), l1Owner);
+
+      // Trying to transfer ownership from non-cross-domain owner should fail.
+      assert(await didContractThrow(depositBox.methods.transferL1Owner(user1).send({ from: rando })));
+
+      // Trying to call correctly via the L2 message impersonator, but from the wrong xDomainMessageSender should revert.
+      l2CrossDomainMessengerMock.smocked.xDomainMessageSender.will.return.with(() => rando);
```

Comment:
this is the only smocked logic done here where we are simply setting the `xDomainMessageSender` will return a particular value. this could easily be done with a solidity mock.

In future, we can check that specific calls `l2CrossDomainMessengerMock.smocked.sendMessage.calls[0]` contain things like `_target` and `_message` to see that specific calls happened to the bridge. this would _not_ be possible with a solidity mock.

Reply:
This is pretty awesome and makes this impersonation very succinct